File: easy/03-draw-gift/main.js
/**
 * @param {number} size - The size of the gift
 * @param {string} symbol - The symbol to draw
 * @returns {string} The gift drawn
 */
function drawGift(size, symbol) {
  if (size < 2) return ''

  // This is top and the border part of the matrix
  const top = symbol.repeat(size)

  // Now this is the pattern fill including the spaces
  const middle = symbol + ' '.repeat(Math.max(0, size - 2)) + symbol

  // Joining all in a single string, top - middle - border
  const lines = [top, ...Array(Math.max(0, size - 2)).fill(middle), top].join(
    '\n'
  )

  return lines
}

File: easy/03-draw-gift/main.py
def draw_gift(size, symbol):
    """
    Dibuja un "regalo" cuadrado de tamaño `size` usando `symbol` como borde.
    Devuelve cadena vacía si size < 2 o si symbol está vacío.
    """
    if size < 2:
        return ''

    sym = symbol[0] if symbol else ''
    if not sym:
        return ''

    top = sym * size
    middle = sym + ' ' * max(0, size - 2) + sym
    lines = [top] + [middle] * max(0, size - 2) + [top]
    return '\n'.join(lines)